### DISCUSSION TITLE
Error message: "Cannot process 0 length data"

### DISCUSSION THREAD

**Discussion by qupig** [keepassxreboot/keepassxc#10764] [Q&A]
I see this error when saving the database:
```
Cannot process 0 length data
```

I'm not sure what it means, when I click the save button again, the error doesn't appear again, but the database title still contains a `*` (modification status) and the save button is disabled at now, I'm not sure if my changes have actually been saved.

Can anyone offer some help with this error message? What does it mean? Is there anything I need to check?

KeePassXC 2.7.8 (f6757d3)

**Comment by droidmonkey** [comment]
that is interesting, did you run out of RAM? This error message occurs from the SymmetricCipher process method and will display if you try to encrypt 0 bytes of data (aka an empty string).

**Reply by qupig** [reply]
No, I have 64GB of RAM and the current actual usage is low.
I'm not sure where there is a `0 byte` data. 
I just modified a few entries and saving database. 
This is a fairly routine operation, nothing fancy.
It's also my first time encountering this error, so I'm nervous.

**Reply by droidmonkey** [reply]
Very strange, unfortunately I can't go any further because this could be triggered from many parts of the code base.

**Reply by qupig** [reply]
Understandably, this is a sporadic issue and cannot be reproduced.
I haven't found any corruption so far, and I keep plenty of backups, but it was good to document the issue here.

**Reply by e00E** [reply]
I had the same issue. Randomly happened for the first time today. I guess there are no logs or anything to check what went wrong?

**Comment by stolendata** [comment]
I experienced this today with 2.7.12 on macOS 14/Arm64.
 
After opening an entry up and changing its password, instead of just pressing the OK button on the right I first pressed Apply on the left and then OK. The error message appeared in a red dialog just after pressing OK. I've tried reproducing it but I cannot. Now I can't help worrying about keychain corruption.
